### PR TITLE
HDDS-8744. Clean up checked functional interfaces

### PR DESCRIPTION
**Pull request by adoroszlai**
## What changes were proposed in this pull request?

Clean up code related to `Checked*` functional interfaces:

 * `CheckedBiFunction` used in `XceiverClientSpi` and `OMVolumeAclRequest` should be `CheckedBiConsumer` (no return value)
 * `CheckedFunction` in `ExecutionUtil` is really a `CheckedRunnable` (no return value)
 * `CheckedBiFunction` is defined in multiple integration tests
 * reuse interfaces from Ratis instead of duplicating them

https://issues.apache.org/jira/browse/HDDS-8744

## How was this patch tested?

https://github.com/adoroszlai/hadoop-ozone/actions/runs/5143653792

### CODE REVIEW

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/TestCSMMetrics.java [hunk 1]
```diff
@@ -132,10 +129,10 @@ static void runContainerStateMachineMetrics(
       assertCounter("WriteChunkMsNumOps", 0L, metric);
       double applyTransactionLatency = getDoubleGauge(
           "ApplyTransactionNsAvgTime", metric);
-      assertTrue(applyTransactionLatency == 0.0);
+      assertEquals(0.0, applyTransactionLatency, 0.0);
```
Comment (Galsza) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/TestCSMMetrics.java:132]: Wouldn't Math.ulp be better for epsilon here? What does 0.0 mean as the epsilon here?

Comment (adoroszlai) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/TestCSMMetrics.java:132]: IDEA simplified the assertion to this one.  I think it's the same as `assertEquals(0.0, applyTransactionLatency)`, without any `delta`.

**Review by szetszwo — APPROVED** [review]
+1 the change looks good.  Thanks for working on this!

**Comment by adoroszlai** [comment]
Thanks @Galsza, @szetszwo for the review.

**Pull request merged**